src/main/scala/com/fulcrumgenomics/bam/Bams.scala

```diff
@@ -98,6 +98,9 @@ case class Template(r1: Option[SamRecord],
     Template(x1, x2)
   }
 
+  /** True if the template has only fragment (R1) reads, false if paired (has both R1 and R2 reads). */
+  def unpaired: Boolean = r2.isEmpty
```

Comment:
This seems dangerous.  What if `r1.paired` is true, but `r2` is missing?  What if somehow we have secondary or supplementary alignments for r2 but the primary is absent?  Perhaps `r1.exists(_.unpaired) && r2.isEmpty`?